### PR TITLE
Fix: Presto change the implementation of set_timezone_to_utc

### PR DESCRIPTION
**Pull request by pppsunil**
# Fix: Presto change the implementation of set_timezone_to_utc

## Problem
I tried running the following data-diff for Presto data source
```
data-diff "presto://user:password@host-name8443/hive/public?auth=basic&http_scheme=https" -k id -l 5 --json test_table_1 test_table_2
```
It failed with the following error. I made changes in `def query_cursor(c, sql_code)` method to print the query that it was trying to execute
```
 query_cursor SET TIME ZONE '+00:00'
21:52:57 ERROR    PrestoUserError(type=USER_ERROR, name=SYNTAX_ERROR, message="line 1:5:   __main__.py:332
                  mismatched input 'TIME'. Expecting: 'ROLE', 'SESSION'",
                  query_id=20231013_045257_12027_sxb6t)
Traceback (most recent call last):
  File "/Users/spatil/study/datadiff/presto/venv/bin/data-diff", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/__main__.py", line 328, in main
    return _data_diff(
           ^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/__main__.py", line 408, in _data_diff
    db1 = connect(database1, threads1 or threads)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/databases/_connect.py", line 276, in __call__
    conn = self.connect_to_uri(db_conf, thread_count, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/databases/_connect.py", line 206, in connect_to_uri
    return self._connection_created(db)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/databases/_connect.py", line 302, in _connection_created
    db.query(db.dialect.set_timezone_to_utc())
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/databases/base.py", line 893, in query
    res = self._query(sql_code)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/databases/presto.py", line 196, in _query
    return query_cursor(c, sql_code)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/data_diff/databases/presto.py", line 42, in query_cursor
    c.execute(sql_code)
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/prestodb/dbapi.py", line 267, in execute
    self._iterator = iter(self._query.execute())
                          ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/prestodb/client.py", line 550, in execute
    self._result._rows += self.fetch()
                          ^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/prestodb/client.py", line 560, in fetch
    status = self._request.process(response)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/spatil/study/datadiff/presto/venv/lib/python3.11/site-packages/prestodb/client.py", line 409, in process
    raise self._process_error(response["error"], response.get("id"))
prestodb.exceptions.PrestoUserError: PrestoUserError(type=USER_ERROR, name=SYNTAX_ERROR, message="line 1:5: mismatched input 'TIME'. Expecting: 'ROLE', 'SESSION'", query_id=20231013_045257_12027_sxb6t)
```

Presto is failing when it  try to execute `SET TIME ZONE '+00:00'` query because as per presto [documentation](https://prestodb.io/docs/current/release/release-0.66.html#:~:text=The%20time%20zone%20of%20the%20session%20can%20be%20set%20on,method%20in%20the%20JDBC%20driver.) 
```
The time zone of the session can be set on a per-query basis using the X-Presto-Time-Zone HTTP header, or via the PrestoConnection.setTimeZoneId(String) method in the JDBC driver
```
## Solution
I changed the implementation of `set_timezone_to_utc()` to
```
    def set_timezone_to_utc(self) -> str:
        raise NotImplementedError()
```

### CODE REVIEW

File: data_diff/databases/presto.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def parse_type(
         return super().parse_type(table_path, col_name, type_repr, datetime_precision, numeric_precision)
 
     def set_timezone_to_utc(self) -> str:
```
Comment (vvkh) [inline data_diff/databases/presto.py:154]: @nolar @pppsunil 
wdyt about making `set_timezone_to_utc` a method of Database class with default implementation like 
`self.query(self.dialect.set_timezone_to_utc())`? 
Then we'll be able to override the implementation for Presto.

Comment (pppsunil) [reply data_diff/databases/presto.py:154]: @vvkh I am trying to understand what you mean 
1. Right now the `set_timezone_to_utc` is an abstract method defined in the `BaseDialect` class like this 
```    @abstractmethod
    def set_timezone_to_utc(self) -> str:
        "Provide SQL for setting the session timezone to UTC"
```
2. Every database has its own implementation of the `Dialect` class and provides an implementation of this method specific to the database Ex
```
    def set_timezone_to_utc(self) -> str:
        return "SET TIME ZONE 'UTC'"
``` 
3. In this PR I am changing the implementation specific to Presto indicating that Presto does not support setting time zone to UTC
4. I realized that Trino database + Dialect classes are implemented by extending Presto and Trino does support setting UTC timezone so made a change to override `set_timezone_to_utc()` method in Trino

I believe this change should only affect Presto. 

About " about making set_timezone_to_utc a method of Database class with default implementation like
self.query(self.dialect.set_timezone_to_utc())?" -> TBH I like the current abstraction which says hey how you set UTC timezone at session level is dialect specific.

Comment (pppsunil) [reply data_diff/databases/presto.py:154]: @vvkh I am trying to understand what you mean 
1. Right now the `set_timezone_to_utc` is an abstract method defined in the `BaseDialect` class like this 
```    @abstractmethod
    def set_timezone_to_utc(self) -> str:
        "Provide SQL for setting the session timezone to UTC"
```
2. Every database has its own implementation of the `Dialect` class and provides implementation of this method specific to the database 
```
    def set_timezone_to_utc(self) -> str:
        return "SET TIME ZONE 'UTC'"
```

Comment (vvkh) [reply data_diff/databases/presto.py:155]: > TBH I like the current abstraction which says hey how you set UTC timezone at session level is dialect specific.
Yes, but there's the assumption that you always execute some kind of SQL query to set timezone. Apparently, that's not always the case, just as with Presto.

That's why I'm thinking if we might need to change the abstraction to allow setting a timezone on the "Database" entity level, not the "SQL dialect" level.

**Pull request closed without merging**